tools/topology/platform/intel/intel-generic-dmic-kwd.m4

```diff
@@ -0,0 +1,121 @@
+#
+# Topology for digital microphones array
+#
+
+include(`platform/intel/dmic.m4')
+
+# variable that need to be defined in upper m4
+# define(CHANNELS, `4')
+
+# define(DMIC_PCM_48k_ID, `99')
+# define(DMIC_PIPELINE_48k_ID, `4')
+# define(DMIC_DAI_LINK_48k_ID, `1')
+
+# define(DMIC_PCM_16k_ID, `7')
+# define(DMIC_PIPELINE_16k_ID, `9')
+# define(DMIC_PIPELINE_KWD_ID, `10')
+# define(DMIC_DAI_LINK_16k_ID, `2')
+
+# define(KWD_PIPE_SCH_DEADLINE_US, 20000)
+
+#
+# Define the pipelines
+#
+
+dnl PIPELINE_PCM_ADD(pipeline,
+dnl     pipe id, pcm, max channels, format,
+dnl     period, priority, core,
+dnl     pcm_min_rate, pcm_max_rate, pipeline_rate,
+dnl     time_domain, sched_comp)
+
+# Passthrough capture pipeline using max channels defined by CHANNELS.
+
+# Set 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_ADD(sof/pipe-passthrough-capture.m4,
+	DMIC_PIPELINE_48k_ID, DMIC_PCM_48k_ID, CHANNELS, s32le,
+	1000, 0, 0, 48000, 48000, 48000)
+
+#
+# KWD configuration
+#
+
+# Passthrough capture pipeline 7 on PCM 7 using max 2 channels.
+# Schedule 20000us deadline on core 0 with priority 0
+PIPELINE_PCM_DAI_ADD(sof/pipe-kfbm-capture.m4,
+	DMIC_PIPELINE_16k_ID, DMIC_PCM_16k_ID, 2, s24le,
+	KWD_PIPE_SCH_DEADLINE_US, 0, 0, DMIC, 1, s32le, 3,
+	16000, 16000, 16000)
+
+
```

Comment:
Can we have the 'PGA8.0 8 KWD Capture Volume' module also added here similar to CML?
That way we can make this more generic?

Reply:
Updated.